src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -208,9 +255,6 @@ DQMatMulGQi::DQMatMulGQi(Context::Ref ctx) {
             matched_qweight->set_partial_shape(tw_shape);
             matched_qweight->validate_and_infer_types();
 
-            matched_qcoeff->set_element_type(ov::element::f16);
```

Comment:
Why did you move it to `to_f16`? I wonder if those rather invasive changes still work as expected for the repeated block?

Reply:
As you can see a couple of lines before, `to_f16` looks like some in-place operation rather than something pure. We have
```c++
ctx.get().to_f16(matched_qcoeff);
```
instead of
```c++
auto new_qcoeff = ctx.get().to_f16(matched_qcoeff);
```

So since we mark the original `matched_qcoeff` parameter to be converted, why don't the `to_f16()` do the change to the parameter element type as well? It must've been done there initially, I only figured it out later.

Don't get your point on invasive changes and repeating blocks.